packages/core/src/utilities/worldToImageCoords.ts

```diff
@@ -59,10 +63,11 @@ function worldToImageCoords(
   ];
 
   if (
-    imageCoords[0] < 0 ||
-    imageCoords[0] >= columns ||
-    imageCoords[1] < 0 ||
-    imageCoords[1] >= rows
+    checkCoordinateSpace &&
+    (imageCoords[0] < 0 ||
+      imageCoords[0] > columns ||
```

Comment:
The image coordinates have to be strictly larger than columns,rows as the last pixel is columns-1...columns inclusive.